src/js/controller/controller.js

```diff
@@ -670,8 +670,8 @@ Object.assign(Controller.prototype, {
                         itemsShown: [ nextUp ],
                         feedData: nextUp.feedData,
                     });
+                    _item(nextUp.index);
```

Comment:
This should work for playlist items where `nextUp.index` equals `jwplayer().getPlaylistIndex() + 1`.

If `nextUp` is a recommendation and not part of the playlist, then `_load(nextUp)` should be called to replace the playlist.

Make sure that `feedData` is also set correctly in this scenario, as well as any other logic that is part of related's `selectItem` "next" action that is exposed publicly (plugin events, etc...).